crates/utils/src/rate_limit/rate_limiter.rs

```diff
@@ -1,15 +1,43 @@
-use crate::IpAddr;
-use std::{collections::HashMap, time::Instant};
-use strum::IntoEnumIterator;
+use enum_map::{enum_map, EnumMap};
+use once_cell::sync::Lazy;
+use std::{
+  collections::HashMap,
+  net::Ipv6Addr,
+  time::{Duration, Instant},
+};
 use tracing::debug;
 
+static START_TIME: Lazy<Instant> = Lazy::new(Instant::now);
+
+#[derive(Debug, Clone, Copy)]
+struct InstantSecs {
+  secs: u32,
+}
```

Comment:
You added this type to use less memory? How much difference does it make compared to Instant? And can you add a comment to document this?